Build failed with -Prelease

- [x] I have searched the [issues](https://github.com/apache/incubator-dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/incubator-dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: master branch
* Operating System version: macos
* Java version: java11

### Steps to reproduce this issue

1. `./mvnw package  -DskipTests -Pjava8-vm-args -Prelease`

### Expected Result

build success

### Actual Result

build failed

If there is an exception, please attach the exception trace:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.1:jar (attach-javadoc) on project dubbo: MavenReportException: Error while generating Javadoc:
[ERROR] Exit code: 1 - /Users/xxx/Documents/incubator-dubbo/dubbo-common/src/main/java/org/apache/dubbo/common/logger/log4j2/Log4j2LoggerAdapter.java:23: error: package org.apache.logging.log4j does not exist
[ERROR] import org.apache.logging.log4j.LogManager;
[ERROR]                                ^
[ERROR] /Users/xxx/Documents/incubator-dubbo/dubbo-common/src/main/java/org/apache/dubbo/common/logger/log4j2/Log4j2LoggerAdapter.java:35: error: package org.apache.logging.log4j does not exist
[ERROR]     private static org.apache.logging.log4j.Level toLog4j2Level(Level level) {
[ERROR]                                            ^
[ERROR] /Users/xxx/Documents/incubator-dubbo/dubbo-common/src/main/java/org/apache/dubbo/common/logger/log4j2/Log4j2LoggerAdapter.java:57: error: package org.apache.logging.log4j does not exist
[ERROR]     private static Level fromLog4j2Level(org.apache.logging.log4j.Level level) {
[ERROR]                                                                  ^
[ERROR] /Users/xxx/Documents/incubator-dubbo/dubbo-common/src/main/java/org/apache/dubbo/common/logger/log4j2/Log4j2Logger.java:23: error: package org.apache.logging.log4j does not exist
[ERROR]     private final org.apache.logging.log4j.Logger logger;
[ERROR]                                           ^
[ERROR] /Users/xxx/Documents/incubator-dubbo/dubbo-common/src/main/java/org/apache/dubbo/common/logger/log4j2/Log4j2Logger.java:25: error: package org.apache.logging.log4j does not exist
[ERROR]     public Log4j2Logger(org.apache.logging.log4j.Logger logger) {
[ERROR]                                                 ^
[ERROR]
[ERROR] Command line was: /Library/Java/JavaVirtualMachines/jdk-11.0.1.jdk/Contents/Home/bin/javadoc @options @packages
[ERROR]
[ERROR] Refer to the generated Javadoc files in '/Users/xxx/Documents/incubator-dubbo/dubbo-all/target/apidocs' dir.
[ERROR]
[ERROR] -> [Help 1]
```
### Solution
`log4j-core` was defined in `dubbo-common/pom.xml` as `provied`
change to default scope `compile`

--- a/pom.xml
+++ b/pom.xml
@@ -276,6 +276,9 @@
         </profile>
         <profile>
             <id>release</id>
+            <properties>
+                <log4j2_version>2.11.1</log4j2_version>
+            </properties>
             <build>
                 <plugins>
                     <plugin>
@@ -289,6 +292,18 @@
                                 </goals>
                                 <configuration>
                                     <doclint>none</doclint>
+                                    <additionalDependencies>
+                                        <additionalDependency>
+                                            <groupId>org.apache.logging.log4j</groupId>
+                                            <artifactId>log4j-api</artifactId>
+                                            <version>${log4j2_version}</version>
+                                        </additionalDependency>
+                                        <additionalDependency>
+                                            <groupId>org.apache.logging.log4j</groupId>
+                                            <artifactId>log4j-core</artifactId>
+                                            <version>${log4j2_version}</version>
+                                        </additionalDependency>
+                                    </additionalDependencies>
                                 </configuration>
                             </execution>
                         </executions>

